app/controllers/address_info_controller.rb

```diff
@@ -3,5 +3,30 @@ def new
   end
 
   def create
+    @address = Address.find_by(user_id: current_user)
+    if @address
+      if @address.update(address_params)
+        flash[:notice] = "Thank You for updating your address"
+      else
+        render 'new'
+      end
+    else
+      @address = Address.new(address_params)
+      @address.user = current_user
+      if @address.save
+        flash[:notice] = "Thank You for providing your address"
+      else
+        render 'new'
+      end
```

Comment:
@AkshayPashnathSuresh too much line of code. code optimization is important. 

```suggestion
   address = Address.where(user_id: current_user).first_or_initialize
    if address.update(address_params)
      flash.now[:notice] = "Thank You for updating your address"
    else
      render 'new'
    end
```